build_tools/cmake/iree_check_test.cmake

```diff
@@ -140,6 +142,72 @@ function(iree_check_test)
   endif()
 
   iree_package_name(_PACKAGE_NAME)
+
+  # If the source file is from the web, create a custom command to download
+  # it and wrap that with a custom target so later we can use for dependency.
+  if("${_RULE_SRC}" MATCHES "^https?://")
+    set(_SRC_URL "${_RULE_SRC}")
+    # Update the source file to the downloaded-to place.
+    string(REPLACE "/" ";" _SRC_URL_SEGMENTS "${_SRC_URL}")
+    list(POP_BACK _SRC_URL_SEGMENTS _LAST_URL_SEGMENT)
+    set(_DOWNLOAD_TARGET "${_PACKAGE_NAME}_${_LAST_URL_SEGMENT}")
+
+    # Strip off gzip/tar suffix if present (downloader unpacks if necessary)
+    string(REGEX REPLACE "(\.gz)|(\.tar\.gz)$" "" _SRC_BASENAME "${_LAST_URL_SEGMENT}")
+    set(_RULE_SRC "${CMAKE_CURRENT_BINARY_DIR}/${_SRC_BASENAME}")
+    string(REGEX REPLACE "(https?://.*)$" ""  _NAME_BASE "${_RULE_NAME}")
+    set(_RULE_NAME "${_NAME_BASE}${_SRC_BASENAME}")
+    if(NOT TARGET "${_DOWNLOAD_TARGET}")
+      add_custom_command(
+        OUTPUT "${_RULE_SRC}"
+        COMMAND
+          "${Python3_EXECUTABLE}" "${IREE_ROOT_DIR}/build_tools/scripts/download_file.py"
+          "${_SRC_URL}" -o "${_RULE_SRC}"
+        DEPENDS
+          "${IREE_ROOT_DIR}/build_tools/scripts/download_file.py"
+        COMMENT "Downloading ${_SRC_URL}"
+      )
+      add_custom_target("${_DOWNLOAD_TARGET}"
+        DEPENDS "${_RULE_SRC}"
+      )
+    endif()
+    set(_MODULE_SOURCE_TARGET "${_DOWNLOAD_TARGET}")
+  endif()
+
+  # If the source is a TFLite file, import it. The iree-tflite-tool must be
+  # found under PATH or specified by the `IREE_IMPORT_TFLITE_BIN` environment
+  # variable.
+  if("${_RULE_SRC}" MATCHES "\.tflite$")
+    if(NOT DEFINED ENV{IREE_IMPORT_TFLITE_BIN})
+      find_program(IREE_IMPORT_TFLITE_BIN
+        "iree-import-tflite" HINTS ENV PATH REQUIRED)
```

Comment:
I'd strongly advise against this approach (for general development / in the core project), for a few reasons. Let's take a step back and discuss more broadly (outside the context of RISC-V testing) before proceeding.

* `iree-import-tflite` does not run on Windows
* Frontends must not be included in core project tests. We need to enforce layering between frameworks, input dialects, and the runtime. We have a few spots in the project that cut across these layers in this way already, _and those need to be removed, not expanded upon_. There may be space for tests using frontends in this way in other repositories like https://github.com/iree-org/iree-samples or a subproject in the core repo, but we should to keep functions like `iree_check_test` simple/pure/focused.
* The usual way to get a binary in our CMake files is `iree_get_executable_path()`, but we already have some different logic for `iree-import-tflite` already: https://github.com/iree-org/iree/blob/2ef36afdb9682d98f347f75a7f8f4bbbf2a7ee6d/CMakeLists.txt#L749-L755
  * We would want a similar mechanism for gracefully disabling certain tests if the import tool is not available here (wherever this sort of test ends up)